### ISSUE TITLE
Please clarify "Strong consistency"

### ISSUE THREAD

**Issue by slowfranklin** [tikv/tikv#13956]
Hi folks!

"Strong consistency" doesn't seem to be a well defined term in the literature on distributed databases, I wonder whether the documentation could be improved to use a more widely used model name?

Eg on [Wikipedia](https://en.wikipedia.org/wiki/Consistency_model)  "Strong consistency" is just a grouping of several models including "Strict consistency", "Sequential consistency" and more. Similarily [Jepsen](https://jepsen.io/consistency) also doesn't know about "Strong consistency".

Can someone please clarify? Thanks a lot!

Cheers!
-slow

**Comment by BusyJay** [comment]
Strong consistency is a relative concept to eventual consistency in distributed system. Here it means after a KV is written, then the KV must be read by any client at any time unless it's updated or deleted (that's also why it's call immediate consistency sometimes). While eventual consistency means after a KV is written, the KV may be read by any client eventually.

As for consistency model, TiKV can provide linearizability if stale read and follower read are not used.

**Comment by slowfranklin** [comment]
Thanks for the clarification! The problem with the term "Strong consistency" is that is actually a spectrum ranging from "Sequential Consistency", "Linearizability" to "Strict Serializability" (with a few more in between). If TiKV supports "Linearizability" as strongest model, that's awesome and should be documented as such to avoid ambiguities. Thanks!

**Comment by dananjayamahesh** [comment]
Hi, I have a follow up question.  When you say 'Linearizability', do you mean single-key operations (like read/write/read-modify-writes)? or does this so-called 'Linearizability' support muti-key transactions as well? because in the readme it says tikv is a transactional KVS. much appreciated. thanks.

**Comment by slowfranklin** [comment]
Linearizability is a consistency model related to single key operations, cf [Jepsen](https://jepsen.io/consistency): everything on the right side of the tree applies to single-key operations. The left side of the tree has the classic consistency models from SQL databases.  So if you need a multi-key consistent distributed database you need Strict Serializability, which combines both properties.

**Comment by dananjayamahesh** [comment]
sure. does tikv/tidb supports strict serializability?. what is the consistency model of multi-key tikv transactions?.  is it strict serializability or something user defined?. tks.

**Comment by slowfranklin** [comment]
Yes, this is a shame and it should be clearly documented what consistency is provided. TiKV looks like a fine product and in [TiKV Doc](https://tikv.org/docs/3.0/reference/faq/#how-does-tikv-compare-to-nosql-databases-like-cassandra-hbase-or-mongodb) it claims 

>  externally consistent distributed transactions

which I'd argue can only be interpreted as implying [Strict serializable](https://jepsen.io/consistency/models/strict-serializable) consistency, which would excellent.

**Comment by nmabhinandan** [comment]
Yes. `externally consistent` would imply strict serializability. Confirming it by having it written somewhere in the docs would be awesome.
